irodsadmin/management/commands/irodsorphans.py

```diff
@@ -25,184 +26,234 @@
 table_builder = SampleSheetTableBuilder()
 
 
-def get_assay_collections(assays, irods_backend):
-    """Return a list of all assay collection names."""
-    return [irods_backend.get_path(a) for a in assays]
-
-
-def get_assay_subcollections(studies, irods_backend):
-    """Return a list of all assay row colletion names."""
-    collections = []
-    for study in studies:
-        try:
-            study_tables = table_builder.get_study_tables(study)
-        except Exception as ex:
-            logger.error(
-                'Study table building exception for "{}" '
-                'in project "{}" ({}): {}'.format(
-                    study.get_display_name(),
-                    study.investigation.project.title,
-                    study.investigation.project.sodar_uuid,
-                    ex,
-                )
+# Local constants
+DELETED = '<DELETED>'
+ERROR = '<ERROR>'
+
+
+class Command(BaseCommand):
+    """Command to find orphans in iRODS collections."""
+
+    help = 'Find orphans in iRODS project collections.'
+
+    def __init__(self):
+        super().__init__()
+        self.irods_backend = get_backend_api('omics_irods')
+
+    def _sort_colls_on_projects(self, collections):
```

Comment:
Sorting collections by projects doesn't seem to be working correctly.

In my dev project, I have two projects with both orphaned sample repository data and landing zone data. One of the projects has been deleted while the other still exists.

In the output below, you can see that the data is still ordered so that landing zones from all projects come first, followed by sample data from all projects. All of this should be sorted so that:

- We first get all from project X
- Then we get all from project Y

Output:
```
6fcdf2b7-b4fe-44d8-a6fe-0634eba54b7d;Test Category / Orphaned Data Project;/sodarZone/projects/6f/6fcdf2b7-b4fe-44d8-a6fe-0634eba54b7d/landing_zones/alice/demo_study/demo_assay/20230818_124813_orphan_zone;0;0 bytes
43fab04b-4bbe-4b1e-9ee9-a9d0dd4abb09;<DELETED>;/sodarZone/projects/43/43fab04b-4bbe-4b1e-9ee9-a9d0dd4abb09;2;40 bytes
43fab04b-4bbe-4b1e-9ee9-a9d0dd4abb09;<DELETED>;/sodarZone/projects/43/43fab04b-4bbe-4b1e-9ee9-a9d0dd4abb09/landing_zones/alice/investigation_title/investigation_title_transcription_profiling_nucleotide_sequencing/20230719_153525_orphaned_zone;1;20 bytes
43fab04b-4bbe-4b1e-9ee9-a9d0dd4abb09;<DELETED>;/sodarZone/projects/43/43fab04b-4bbe-4b1e-9ee9-a9d0dd4abb09/sample_data/study_956c34ad-4f54-4770-9fa5-0d8c720336e5;1;20 bytes
43fab04b-4bbe-4b1e-9ee9-a9d0dd4abb09;<DELETED>;/sodarZone/projects/43/43fab04b-4bbe-4b1e-9ee9-a9d0dd4abb09/sample_data/study_956c34ad-4f54-4770-9fa5-0d8c720336e5/assay_a333c690-c95e-4a1e-be0e-2f7bef5c2615;1;20 bytes
6fcdf2b7-b4fe-44d8-a6fe-0634eba54b7d;Test Category / Orphaned Data Project;/sodarZone/projects/6f/6fcdf2b7-b4fe-44d8-a6fe-0634eba54b7d/sample_data/study_845d4d27-47d9-4273-9fea-62cf55b5de79/assay_ea75df18-2911-456b-8412-06bb1df4607d/orphaned_collection;2;45 bytes
```